src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,804 @@
+            A model property specifying whether the predicted depth maps are scale-invariant. This value must be set in
+            the model config. When used together with the `shift_invariant=True` flag, the model is also called
+            "affine-invariant". NB: overriding this value is not supported.
+        shift_invariant (`bool`, *optional*):
+            A model property specifying whether the predicted depth maps are shift-invariant. This value must be set in
+            the model config. When used together with the `scale_invariant=True` flag, the model is also called
+            "affine-invariant". NB: overriding this value is not supported.
+        default_denoising_steps (`int`, *optional*):
+            The minimum number of denoising diffusion steps that are required to produce a prediction of reasonable
+            quality with the given model. This value must be set in the model config. When the pipeline is called
+            without explicitly setting `num_inference_steps`, the default value is used. This is required to ensure
+            reasonable results with various model flavors compatible with the pipeline, such as those relying on very
+            short denoising schedules (`LCMScheduler`) and those with full diffusion schedules (`DDIMScheduler`).
+        default_processing_resolution (`int`, *optional*):
+            The recommended value of the `processing_resolution` parameter of the pipeline. This value must be set in
+            the model config. When the pipeline is called without explicitly setting `processing_resolution`, the
+            default value is used. This is required to ensure reasonable results with various model flavors trained
+            with varying optimal processing resolution values.
+    """
+
+    model_cpu_offload_seq = "text_encoder->unet->vae"
+    supported_prediction_types = ("depth", "disparity")
+
+    def __init__(
+        self,
+        unet: UNet2DConditionModel,
+        vae: AutoencoderKL,
+        scheduler: Union[DDIMScheduler, LCMScheduler],
+        text_encoder: CLIPTextModel,
+        tokenizer: CLIPTokenizer,
+        prediction_type: Optional[str] = None,
+        scale_invariant: Optional[bool] = True,
+        shift_invariant: Optional[bool] = True,
+        default_denoising_steps: Optional[int] = None,
+        default_processing_resolution: Optional[int] = None,
+    ):
+        super().__init__()
+
+        if prediction_type not in self.supported_prediction_types:
+            logger.warning(
+                f"Potentially unsupported `prediction_type='{prediction_type}'`; values supported by the pipeline: "
+                f"{self.supported_prediction_types}."
+            )
+
+        self.register_modules(
+            unet=unet,
+            vae=vae,
+            scheduler=scheduler,
+            text_encoder=text_encoder,
+            tokenizer=tokenizer,
+        )
+        self.register_to_config(
+            prediction_type=prediction_type,
```

Comment:
I think `prediction_type` is not used?

Reply:
It is used in the check above and is characteristic of the predicted modality

Reply:
I still don't think this is needed -
it is not used anywhere in the code, no? e.g. would the results be any difference between `prediction_type="depth"`, and `prediction_type="disparity"? or even `prediction_type=None`?

Reply:
What would you recommend doing with this field if we decided to specify the predicted modality type in the config.json of the models? I thought anything that goes into __init__ should also be mirrored in `register_to_config` to ensure smooth load-store of the checkpoints... Like for example, here: https://huggingface.co/prs-eth/marigold-disparity-affine-v0-1/blob/main/model_index.json (verbatim below)

```json
{
    "_class_name":"MarigoldPipeline",
    "_diffusers_version":"0.24.0",
    "prediction_type": "disparity",
    "scale_invariant": true,
    "shift_invariant": true, 
    "unet":[
       "diffusers",
       "UNet2DConditionModel"
    ],
    "vae":[
       "diffusers",
       "AutoencoderKL"
    ],
    "scheduler":[
       "diffusers",
       "DDIMScheduler"
    ],
    "text_encoder":[
       "transformers",
       "CLIPTextModel"
    ],
    "tokenizer":[
       "transformers",
       "CLIPTokenizer"
    ]
 }
```

 To your question, yes, currently the code is identical for the two modality types.